src/base.jl

```diff
@@ -1,30 +1,83 @@
 # Implementation of GeoInterface for Base Types
 
+
+const PointTuple2 = Tuple{<:Real,<:Real}
+const PointTuple3 = Tuple{<:Real,<:Real,<:Real}
+const PointTuple4 = Tuple{<:Real,<:Real,<:Real,<:Real}
+const PointTuple = Union{PointTuple2,PointTuple3,PointTuple4}
+
 GeoInterface.isgeometry(::Type{<:AbstractVector{<:Real}}) = true
 GeoInterface.geomtrait(::AbstractVector{<:Real}) = PointTrait()
 GeoInterface.ncoord(::PointTrait, geom::AbstractVector{<:Real}) = Base.length(geom)
 GeoInterface.getcoord(::PointTrait, geom::AbstractVector{<:Real}, i) = getindex(geom, i)
 
-GeoInterface.isgeometry(::Type{<:NTuple{N,<:Real}}) where {N} = true
-GeoInterface.geomtrait(::NTuple{N,<:Real}) where {N} = PointTrait()
-GeoInterface.ncoord(::PointTrait, geom::NTuple{N,<:Real}) where {N} = N
-GeoInterface.getcoord(::PointTrait, geom::NTuple{N,<:Real}, i) where {N} = getindex(geom, i)
-
-for i in 2:4
-    sig = NamedTuple{default_coord_names[1:i],NTuple{i,T}} where {T<:Real}
-    GeoInterface.isgeometry(::Type{<:sig}) = true
-    GeoInterface.geomtrait(::sig) = PointTrait()
-    GeoInterface.ncoord(::PointTrait, geom::sig) = i
-    GeoInterface.getcoord(::PointTrait, geom::sig, i) = getindex(geom, i)
+GeoInterface.isgeometry(::Type{<:PointTuple}) = true
+GeoInterface.geomtrait(::PointTuple) = PointTrait()
+GeoInterface.ncoord(::PointTrait, geom::PointTuple) = Base.length(geom)
+GeoInterface.getcoord(::PointTrait, geom::PointTuple, i) = getindex(geom, i)
+
+for (i, pointtype) in enumerate((PointTuple2, PointTuple3, PointTuple4))
+    keys = default_coord_names[1:i+1]
+    sig = NamedTuple{keys,<:pointtype}
+    @eval GeoInterface.isgeometry(::Type{<:$sig}) = true
+    @eval GeoInterface.geomtrait(::$sig) = PointTrait()
+    @eval GeoInterface.ncoord(::PointTrait, geom::$sig) = $i + 1
+    @eval GeoInterface.getcoord(::PointTrait, geom::$sig, i) = getindex(geom, i)
 end
 
-# Custom coordinate order/names NamedTuple
-GeoInterface.isgeometry(::Type{<:NamedTuple{Keys,NTuple{N,T}}}) where {Keys,N,T<:Real} = all(in(default_coord_names), Keys)
-GeoInterface.geomtrait(::NamedTuple{Keys,NTuple{N,T}}) where {Keys,N,T<:Real} = PointTrait()
-GeoInterface.ncoord(::PointTrait, geom::NamedTuple{Keys,NTuple{N,T}}) where {Keys,N,T<:Real} = Base.length(geom)
-GeoInterface.getcoord(::PointTrait, geom::NamedTuple{Keys,NTuple{N,T}}, i) where {Keys,N,T<:Real} = getindex(geom, i)
-GeoInterface.coordnames(::PointTrait, geom::NamedTuple{Keys,NTuple{N,T}}) where {Keys,N,T<:Real} = Keys
 
+const NamedTuplePoints = Union{
```

Comment:
Do you know if there are any downsides to using such large unions? Is the compiler happy with them? In most cases / WKT this would be the order, though it's nice to support swapped X and Y in at least some cases.

```
XY
XYZ
XYM
XYZM
```

Reply:
Being a NamedTuple people want to write it however they want and it's weird when e.g.
`(Y=1, X=2)` doesn't work for people - see the linked issue. Then, if one like that works it seems arbitrary and confusing if any other combination doesn't work.

So I figured just have them all. I imagine there is a tiny bit more compilation time with the union, but it's probably negligible? Maybe see if package load time is any longer? I doubt it will be measurable.